prow/gerrit/adapter/adapter.go

```diff
@@ -73,25 +84,49 @@ var gerritMetrics = struct {
 		// Hopefully this isn't excessive enough to cause metric scraping issues.
 		"repo",
 	}),
+	triggerHelpLatency: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "gerrit_trigger_help_latency",
+		Help:    "Histogram of seconds between triggering event (help) and ProwJob creation time.",
+		Buckets: []float64{5, 10, 20, 30, 60, 120, 180, 300, 600, 1200},
+	}, []string{
+		"org",
+	}),
+	processSingleChangeDuration: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "gerrit_process_single_change_duration",
+		Help:    "Histogram of seconds spent processing a single gerrit change, by instance and repo.",
+		Buckets: []float64{5, 10, 20, 30, 60, 120, 180, 300, 600, 1200},
+	}, []string{
+		"org",
+		"repo",
+	}),
 	changeProcessDuration: prometheus.NewHistogramVec(prometheus.HistogramOpts{
 		Name:    "gerrit_instance_process_duration",
-		Help:    "Histogram of seconds spent processing a single gerrit instance.",
+		Help:    "Histogram of seconds spent processing changes, by instance and repo. Includes gerrit_repo_query_latency and gerrit_instance_change_sync_duration.",
 		Buckets: []float64{5, 10, 20, 30, 60, 120, 180, 300, 600, 1200, 3600},
 	}, []string{
-		"org",
+		"org", "repo",
 	}),
 	changeSyncDuration: prometheus.NewHistogramVec(prometheus.HistogramOpts{
 		Name:    "gerrit_instance_change_sync_duration",
```

Comment:
This sync duration metric and the process duration metric seem to be named backwards.
I think sync duration should cover the entire time we took to do the sync for a repo including the query time, not just the processing time after querying.
Similarly, the processing duration for a repo should just be the time to process after the query has completed.

Can we rectify this in this PR please?

Reply:
To restate, "sync time" == "querying time" + "processing time" (and maybe whatever else we do). SGTM, will update.

Reply:
Yeah sync time is the overall sync loop (querying, processing, everything). Processing time should be the time to complete the sync loop **after** the query completes. Ideally `query time + processing time = sync time`.